src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -2,28 +2,29 @@
 {
     using System;
     using System.Collections.Generic;
-    using Microsoft.ApplicationInsights.AspNetCore.Extensions;
     using Microsoft.ApplicationInsights;
     using Microsoft.ApplicationInsights.AspNetCore;
+    using Microsoft.ApplicationInsights.AspNetCore.DiagnosticListeners;
+    using Microsoft.ApplicationInsights.AspNetCore.Extensions;
     using Microsoft.ApplicationInsights.AspNetCore.TelemetryInitializers;
-    using Microsoft.ApplicationInsights.Channel;
-    using Microsoft.ApplicationInsights.DataContracts;
     using Microsoft.ApplicationInsights.Extensibility;
-    using Microsoft.AspNetCore.Builder;
+    using Microsoft.AspNetCore.Hosting;
+    using Microsoft.AspNetCore.Http;
     using Microsoft.Extensions.Configuration;
     using Microsoft.Extensions.Configuration.Memory;
-    using AspNetCore.Http;
-    using AspNetCore.Mvc.Infrastructure;
+    using Microsoft.Extensions.Options;
 
 #if NET451
-    using ApplicationInsights.DependencyCollector;
-    using ApplicationInsights.Extensibility.PerfCounterCollector;
-    using ApplicationInsights.Extensibility.PerfCounterCollector.QuickPulse;
-    using ApplicationInsights.WindowsServer.TelemetryChannel;
+    using Microsoft.ApplicationInsights.DependencyCollector;
+    using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector;
 #endif
 
+    /// <summary>
+    /// Extension methods for <see cref="IServiceCollection"/> that allow adding Application Insights services to application.
+    /// </summary>
     public static class ApplicationInsightsExtensions
     {
+        private const string VersionKeyFromConfig = "version";
```

Comment:
By default we set it to entry assembly version(https://github.com/Microsoft/ApplicationInsights-aspnetcore/pull/281/files/b650efa55b39cb04fb32c6874812a6f0ba3d08ed#diff-3195debb9b9d179b697b76ff29f8a286R20), but still alow reading from config like the previous version did.